### PR TITLE
(low priority) core,netty,interop-testing: stabilize maxInboundMessageSize API

### PR DESCRIPTION
**Pull request by zpencer**
On server side, `maxMessageSize` is deprecated for
`maxInboundMessageSize` to match the channel builder.

Update usages to use new setter.

fixes https://github.com/grpc/grpc-java/issues/2563

### CODE REVIEW

File: core/src/main/java/io/grpc/ManagedChannelBuilder.java [hunk 1]
```diff
@@ -293,13 +293,12 @@ public T enableFullStreamDecompression() {
    * <p>This method is advisory, and implementations may decide to not enforce this.  Currently,
    * the only known transport to not enforce this is {@code InProcessTransport}.
    *
-   * @param max the maximum number of bytes a single message can be.
-   * @throws IllegalArgumentException if max is negative.
+   * @param bytes the maximum number of bytes a single message can be.
    * @return this
+   * @throws IllegalArgumentException if max is negative.
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ManagedChannelBuilder.java]: s/max/bytes/

File: core/src/main/java/io/grpc/ServerBuilder.java [hunk 2]
```diff
@@ -203,6 +203,25 @@ public T handshakeTimeout(long timeout, TimeUnit unit) {
...
+   * defaults to 4 MiB. The default provides protection to servers who haven't considered the
+   * possibility of receiving large messages while trying to be large enough to not be hit in normal
+   * usage.
+   *
+   * <p>This method is advisory, and implementations may decide to not enforce this.  Currently,
+   * the only known transport to not enforce this is {@code InProcessServer}.
+   *
+   * @param bytes the maximum number of bytes a single message can be.
+   * @return this
+   * @throws IllegalArgumentException if max is negative.
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ServerBuilder.java]: s/max/bytes/

File: core/src/main/java/io/grpc/ManagedChannelBuilder.java [hunk 3]
```diff
@@ -293,13 +293,12 @@ public T enableFullStreamDecompression() {
...
    * @since 1.1.0
    */
-  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2307")
-  public T maxInboundMessageSize(int max) {
+  public T maxInboundMessageSize(int bytes) {
     // intentional nop
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ManagedChannelBuilder.java]: Probably should add in a checkArgument for bytes >= 0.

File: core/src/main/java/io/grpc/ServerBuilder.java [hunk 4]
```diff
@@ -203,6 +203,25 @@ public T handshakeTimeout(long timeout, TimeUnit unit) {
...
+   * @throws UnsupportedOperationException if unsupported.
+   * @since 1.13.0
+   */
+  public T maxInboundMessageSize(int bytes) {
+    throw new UnsupportedOperationException();
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ServerBuilder.java]: Client-side doesn't throw. Since this is advisory, that seems the better approach?

Comment (zpencer) [reply core/src/main/java/io/grpc/ServerBuilder.java]: Changed and added comment.

File: core/src/main/java/io/grpc/ManagedChannelBuilder.java (hunk 3 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/ManagedChannelBuilder.java]: Done

File: core/src/main/java/io/grpc/ServerBuilder.java (hunk 2 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/ServerBuilder.java]: Done

File: core/src/main/java/io/grpc/ManagedChannelBuilder.java (hunk 1 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/ManagedChannelBuilder.java]: Done

File: core/src/main/java/io/grpc/ManagedChannelBuilder.java [hunk 5]
```diff
@@ -293,14 +294,14 @@ public T enableFullStreamDecompression() {
...
+   * @param bytes the maximum number of bytes a single message can be.
    * @return this
+   * @throws IllegalArgumentException if bytes is negative.
    * @since 1.1.0
    */
-  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2307")
-  public T maxInboundMessageSize(int max) {
-    // intentional nop
+  public T maxInboundMessageSize(int bytes) {
+    // intentional noop rather than throw, this method is only advisory.
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/ManagedChannelBuilder.java]: I think you need to have both methods for a short time.  We typically have one release with deprecation before removing it.

Also, this makes it easier to make internal users update before removal.

Comment (ejona86) [reply core/src/main/java/io/grpc/ManagedChannelBuilder.java]: No methods were removed in this PR.

Comment (zpencer) [reply core/src/main/java/io/grpc/ManagedChannelBuilder.java]: Yes, I just renamed `max` to `bytes` to be consistent.

**Pull request merged**